### PR TITLE
404 page && strict mode: true

### PR DESCRIPTION
**Pull request by jota12x**
404 page and strict mode true on typescript.
imagen

### CODE REVIEW

File: src/components/pages/SearchResultsPage.tsx [hunk 1]
```diff
@@ -7,35 +7,35 @@ import queryString from 'query-string';
...
 
 import Layout from 'components/atoms/Layout';
 import SearchHeader from 'components/organisms/SearchHeader';
 import SearchQuery from 'interfaces/SearchQuery';
-import SearchResult from 'interfaces/SearchResult';
+import ISearchResult from 'interfaces/SearchResult';
 import UrlParams from 'interfaces/UrlParams';
 
-const SearchResultsPage = ({ location }) => {
+const SearchResultsPage = ({ location }: { location: any }) => {
```
Comment (rhezaharliman) [inline src/components/pages/SearchResultsPage.tsx]: I'm not sure, but can we make this `any` stricter? Defining a more primitive data type might work if possible.

File: .eslintrc.js [hunk 2]
```diff
@@ -30,16 +30,20 @@ module.exports = {
...
-    "no-irregular-whitespace": ["error", {
-      "skipRegExps": true
-    }],
+    '@typescript-eslint/camelcase': ['off'],
+    '@typescript-eslint/explicit-function-return-type': ['off'],
+    '@typescript-eslint/explicit-member-accessibility': ['off'],
+    '@typescript-eslint/no-empty-interface': ['off'],
+    '@typescript-eslint/no-explicit-any': ['off'],
+    '@typescript-eslint/no-object-literal-type-assertion': ['off'],
+    '@typescript-eslint/prefer-interface': ['off'],
```
[the remark this answers is not in the document]
Comment (jota12x) [reply .eslintrc.js]: Oh, this was default configuration

File: src/App.tsx [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-  return (
-    <Router history={history}>
-      <Switch>
-        <Route exact path="/" component={HomePage} />
-        <Route path="/search" component={SearchResultsPage} />
-      </Switch>
-    </Router>
-  );
-};
+const App = () => (
```
[the remark this answers is not in the document]
Comment (jota12x) [reply src/App.tsx]: Fixed

File: src/components/pages/SearchResultsPage.tsx (hunk 1 above)
Comment (jota12x) [reply src/components/pages/SearchResultsPage.tsx]: hahaha true, sorry, I will fixed.

File: .eslintrc.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rhezaharliman) [reply .eslintrc.js]: Noted!

File: src/components/pages/SearchResultsPage.tsx [hunk 4]
```diff
@@ -1,41 +1,42 @@
...
-  );
+  const [searchResults, setSearchResults] = useState<
+    Array<ISearchResult> | any
+  >([]);
   const [intermediateResults, setIntermediateResults] = useState<
-    Array<SearchResult> | any
+    Array<ISearchResult> | any
+  >([]);
+  const [resultsByPage, setResultsByPage] = useState<
+    Array<ISearchResult> | any
```
[the remark this answers is not in the document]
Comment (jota12x) [reply src/components/pages/SearchResultsPage.tsx]: This has been solved ;)

**Pull request closed without merging**